src/oods.cairo

```diff
@@ -0,0 +1,34 @@
+use cairo_verifier::air::composition::eval_composition_polynomial;
+use cairo_verifier::air::global_values::InteractionElements;
+use cairo_verifier::air::public_input::PublicInput;
+
+struct OodsValues {
+    mask_values: Array<felt252>,
+    split_polynomials: Array<felt252>
+}
+
+fn verify_oods(
+    oods: OodsValues,
+    interaction_elements: InteractionElements,
+    public_input: PublicInput,
+    constraint_coefficients: Array<felt252>,
+    oods_point: felt252,
+    trace_domain_size: felt252,
+    trace_generator: felt252
+) {
+    let composition_from_trace = eval_composition_polynomial(
+        interaction_elements,
+        public_input,
+        oods.mask_values,
+        constraint_coefficients,
+        oods_point,
+        trace_domain_size,
+        trace_generator
+    );
+
+    // TODO support degree > 2?
```

Comment:
This verification is currently only implemented for constraint degree 2.
We can change oods.split_polynomials to simpler type than array, maybe a tuple or dedicated struct.

Reply:
Yes I agree, it's actually not hard to support degree n.
Keep in mind that this is not the degree of the constraints themselves (or the composition polynomial) but rather the number of columns in the composition polynomial trace (or the number of times you "split" the composition polynomial to commit to it).

The reason why I hardcoded degree 2 here is because the cairo0 verifier is doing so too, it seems to be the default when using stone.